File: Sample/CSharp/Sample/Code/TableTest.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Commons;
using Scorpio.Table;
namespace ScorpioProtoTest {
public class TableTest : ITable {
	const string FILE_MD5_CODE = "6002b60ed2ddc12ffecf09c9435044d2";
    private int m_count = 0;
    private Dictionary<int, DataTest> m_dataArray = new Dictionary<int, DataTest>();
    public TableTest Initialize(TableManager tableManager, string fileName) {
        m_dataArray.Clear();
        ScorpioReader reader = new ScorpioReader(TableUtil.GetBuffer(fileName));
        int iRow = TableUtil.ReadHead(reader, fileName, FILE_MD5_CODE);
        for (int i = 0; i < iRow; ++i) {
            DataTest pData = DataTest.Read(tableManager, fileName, reader);
            if (Contains(pData.ID()))
                throw new System.Exception("文件[" + fileName + "]有重复项 ID : " + pData.ID());
            m_dataArray.Add(pData.ID(), pData);
        }
        m_count = m_dataArray.Count;
        reader.Close();
        return this;
    }
    public DataTest GetElement(int ID) {
        if (Contains(ID)) return m_dataArray[ID];
        TableUtil.Warning("DataTest key is not exist " + ID);
        return null;
    }
    public override IData GetValue(int ID) {
        return GetElement(ID);
    }
    public override bool Contains(int ID) {
        return m_dataArray.ContainsKey(ID);
    }
    public override int Count() {
        return m_count;
    }
    public Dictionary<int, DataTest> Datas() {
        return m_dataArray;
    }
}
}

File: ScorpioProto/Java/src/Datas/TableTest.java
package Datas;
//本文件为自动生成，请不要手动修改
import java.util.*;
import ScorpioProto.Commons.*;
import ScorpioProto.Table.*;

public class TableTest implements ITable<Integer, DataTest> {
    final String FILE_MD5_CODE = "f07d3fff17de6b37025a951b272f6c4c";
    private int m_count = 0;
    private HashMap<Integer, DataTest> m_dataArray = new HashMap<Integer, DataTest>();
    public TableTest Initialize(String fileName, IScorpioReader reader) throws Exception {
        int iRow = reader.ReadHead(fileName, FILE_MD5_CODE);
        for (int i = 0; i < iRow; ++i) {
            DataTest pData = DataTest.Read(fileName, reader);
            if (m_dataArray.containsKey(pData.ID()))
                m_dataArray.get(pData.ID()).Set(pData);
            else
                m_dataArray.put(pData.ID(), pData);
        }
        m_count = m_dataArray.size();
        return this;
    }
    public DataTest GetValue(Integer ID) throws Exception {
        if (m_dataArray.containsKey(ID)) 
            return m_dataArray.get(ID);
        throw new Exception("TableTest not found data : " + ID);
    }
    public boolean Contains(Integer ID) {
        return m_dataArray.containsKey(ID);
    }
    public final HashMap<Integer, DataTest> Datas() {
        return m_dataArray;
    }
    public IData GetValueObject(Object ID) throws Exception {
        return GetValue((Integer)ID);
    }
    public boolean ContainsObject(Object ID) {
        return Contains((Integer)ID);
    }
    public int Count() {
        return m_count;
    }
}